src/preferences/dialog/dlgpreflibrary.cpp

```diff
@@ -595,11 +595,38 @@ void DlgPrefLibrary::slotRowHeightValueChanged(int height) {
 }
 
 void DlgPrefLibrary::setLibraryFont(const QFont& font) {
-    lineEdit_library_font->setText(
-            QString("%1 %2 %3pt")
-                    .arg(font.family(),
-                            font.styleName(),
-                            QString::number(font.pointSizeF())));
+    // Update the font name/style/size display
+    QString fontDescription = font.family();
+    const QString style = font.styleName();
+    if (!style.isEmpty()) {
+        // Use the style name if available, likely it's translated
+        fontDescription += ' ' + style;
+    } else {
+        // Else we compose the "style" string from weight and italic.
+        // It's not possible to access the translations used in QFontDialog,
+        // but let's can add them to the user translations
+        const auto weight = font.weight();
+        if (weight >= QFont::Bold) {
+            if (weight >= QFont::Black)
+                fontDescription += ' ' + tr("Black");
+            else if (weight >= QFont::ExtraBold)
+                fontDescription += ' ' + tr("Extra Bold");
+            else if (weight >= QFont::Bold)
+                fontDescription += ' ' + tr("Bold");
+        } else if (weight >= QFont::DemiBold) {
+            fontDescription += ' ' + tr("Semibold");
```

Comment:
To match "Extra Bold." 
```suggestion
            fontDescription += ' ' + tr("Semi Bold");
```